activemodel/lib/active_model/validations/numericality.rb

```diff
@@ -79,18 +80,24 @@ def validate_each(record, attr_name, value)
     private
 
       def is_number?(raw_value)
-        !parse_raw_value_as_a_number(raw_value).nil?
+        !parse_as_number(raw_value).nil?
       rescue ArgumentError, TypeError
         false
       end
 
-      def parse_raw_value_as_a_number(raw_value)
+      def parse_as_number(raw_value)
+        return raw_value.to_d if raw_value.is_a?(Float)
```

Comment:
Too many `return`s here. I'd go with a if/else so it is explicit all the branches.